cdap-api/src/main/java/co/cask/cdap/internal/workflow/DefaultWorkflowSpecification.java

```diff
@@ -32,37 +30,27 @@
  */
 public final class DefaultWorkflowSpecification implements WorkflowSpecification {
```

Comment:
I think now you can get rid of this class and turn `WorkflowSpecification` into a concrete class (Just like `ServiceSpecification`). Also, make sure no guava usage for classes in cdap-api (it's related to https://issues.cask.co/browse/CDAP-20).